packages/ui/src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,189 @@
+     on peut lui passer un booléen mais celui ci doit être accompagné d'un setter (setIsOpen)
+     afin de pouvoir la fermer */
+  isOpen?: boolean;
+  /* accompagne "isOpen" afin de pouvoir fermer la modale */
+  setIsOpen?: (opened: boolean) => void;
+  /** fonction appelée lors de la fermeture de la modale */
+  onClose?: () => void;
+  /** max-width prédéfinies dans le DSFR, valeur par défaut "md" */
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+  /** désactive la fermeture lors du clic sur le fond */
+  disableDismiss?: boolean;
+  /** n'affiche pas le bouton Fermer */
+  noCloseButton?: boolean;
+  /** z-index de la modale, est placé sur l'overlay */
+  zIndex?: string | number;
+};
+
+/*
+ * Basic modal
+ * If you provide isOpen you must add the corresponding setState function.
+ */
+const Modal = ({
+  render,
+  children,
+  title,
+  description,
+  textCenter = true,
+  isOpen,
+  setIsOpen,
+  onClose,
+  size = 'md',
+  disableDismiss,
+  noCloseButton,
+  zIndex,
+}: ModalProps) => {
+  const [open, setOpen] = useState(false);
+
+  const handleOpenChange = () => {
+    setIsOpen ? setIsOpen(!isOpen) : setOpen(!open);
+    if (isOpen || open) {
+      onClose && onClose();
+    }
+  };
+
+  const {refs, context} = useFloating({
+    open: isOpen ?? open,
+    onOpenChange: handleOpenChange,
+  });
+
+  const id = useId();
+  const labelId = `${id}-label`;
+  const descriptionId = `${id}-description`;
+
+  const {getReferenceProps, getFloatingProps} = useInteractions([
+    useClick(context),
+    useRole(context),
+    useDismiss(context, {enabled: !disableDismiss}),
+  ]);
+
+  return (
+    <>
+      {children &&
+        cloneElement(
+          children,
+          getReferenceProps({ref: refs.setReference, ...children.props})
+        )}
+      <FloatingPortal>
+        {(open || isOpen) && (
+          <FloatingOverlay
+            lockScroll
+            style={{
+              display: 'grid',
+              placeItems: 'center',
+              background: preset.theme.extend.colors.overlay,
+              zIndex: zIndex ?? 1100,
+            }}
+          >
+            <FloatingFocusManager context={context}>
+              <div
+                data-test="Modal"
+                {...getFloatingProps({
+                  ref: refs.setFloating,
+                  'aria-labelledby': labelId,
+                  'aria-describedby': descriptionId,
+                  className: classNames(
+                    `
+                      relative flex flex-col self-end gap-8 w-full mt-8 mx-auto p-8
+                      rounded-xl bg-white border border-grey-4 shadow-[0_4px_20px_0px_rgba(0,0,0,0.05)]
+                      sm:self-center sm:w-[calc(100%-3rem)] sm:my-6
+                      md:p-[4.5rem]
+                    `,
+                    sizeToClass[size]
+                  ),
+                })}
+              >
+                {!noCloseButton && (
+                  <button
```

Comment:
Ça serait pas mal de récupérer le code sur ma branche pour utiliser un `<Button/>` ici, avec le variant grey (ou à mettre à jour une fois ma branche en prod).

Reply:
Yes très chaud je me l'étais noté